### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by shaurya-src**

### CODE REVIEW

**Review by Akanksha-nuclei — CHANGES_REQUESTED** [review]

File: Assignment-2/lib/user_details.dart [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    String address;
+    int? rollNumber;
+    List<String>? courses;
+
+    name = Util.inputName();
+    age = Util.inputAge();
+    address = Util.inputAddress();
+    rollNumber = Util.inputRollNumber();
+    courses = Util.inputCourses();
+
```
Comment (Akanksha-nuclei) [inline Assignment-2/lib/user_details.dart:22]: check if you can make the variables private

File: Assignment-2/lib/user.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  static const String _seperator = "__________________________________________________________";
+
+  // Data structure to store students
+  static HashSet registeredRollNumbers = new HashSet<int>();
+  static List<User> registeredUsers = [];
+
+  // Constant constructur to initialize a User object
+  // All the arguments are mandatory
+  User(this.name, this.age, this.address, this.rollNumber, this.courses) {
+    registeredRollNumbers.add(this.rollNumber);
```
[the remark this answers is not in the document]
Comment (shaurya-src) [reply Assignment-2/lib/user.dart]: Removed registeredRollNumbers variable

File: Assignment-2/assets/user_data.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+{"name":"Ankit","age":22,"address":"Mumbai","rollNumber":3,"courses":["A","B","E","F"]}
```
[the remark this answers is not in the document]
Comment (shaurya-src) [reply Assignment-2/assets/user_data.txt]: Implemented encodig

File: Assignment-2/lib/util.dart [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    return double.tryParse(s) != null;
+  }
+
+  // Function to check if number is greater than zero
+  static bool isGreaterThanZero(num number) {
+    return number > 0;
+  }
+
+  // Input user's Full Name
+  static String inputName() {
```
[the remark this answers is not in the document]
Comment (shaurya-src) [reply Assignment-2/lib/util.dart:40]: Re-organized project structure

**Pull request merged**